server/helpers/ffmpeg-utils.ts

```diff
@@ -1,10 +1,47 @@
+import * as fs from 'fs'
 import * as ffmpeg from 'fluent-ffmpeg'
 import { join } from 'path'
 import { VideoResolution } from '../../shared/models/videos'
-import { CONFIG, VIDEO_TRANSCODING_FPS } from '../initializers'
+import { CONFIG, VIDEO_TRANSCODING_FPS, FFMPEG_NICE } from '../initializers'
 import { unlinkPromise } from './core-utils'
 import { processImage } from './image-utils'
 import { logger } from './logger'
+import { checkFFmpegEncoders } from '../initializers/checker'
+
+async function isVideoFileStreamable (path: string): Promise<{ [key: string]: boolean }> {
+  const fileSizeInBytes = fs.statSync(path).size
+  const fileSizeToRead = Math.max(10000, fileSizeInBytes * 0.5)
+  const offset = Math.min(fileSizeToRead, fileSizeInBytes)
+  // basically we want to look at the first 5% of the file, with a minimum of 10000 bytes (if the file is big enough)
+  let stream = fs.createReadStream(path, {
+    encoding: null,
+    start: 0,
+    end: offset
+  })
+
+  const isAVC = await getVideoFileStream(path).then(s => {
+    return s['codec_name'] === 'h264' &&
+           s['bit_rate'] < 3000000
+  })
+  const isAAC = await audio.get(ffmpeg, path).then(s => {
+    return s['codec_name'] === 'aac'
+  })
+  console.log('got it until aac')
+  const hasFastStart = await new Promise<boolean>((res, rej) => {
+    ffmpeg(path, { stdoutLines: 0 })
```

Comment:
While this function is quite optional in the PR scope, it is interesting for the future and essentially bears the same problems as the other Promise-based functions (albeit relying on a call to `ffmpeg` instead of `ffprobe`).

It essentiall tries to mimick the command advised [here in the FFmpeg mailing list](http://ffmpeg.org/pipermail/ffmpeg-devel/2018-July/232693.html): `ffmpeg -i subfile,,start,0,end,10000,,:in.mp4 -v 56 2>&1 | grep -e 
"type:'moov'" -e "type:'mdat'" | head -1 | grep moov`.

To debug after the other functions.

Reply:
I don't see the point of having a function we are not using right now